source/mir/ndslice/topology.d

```diff
@@ -3067,3 +3069,411 @@ unittest
+    static immutable shape435 = [4, 3, 5];
+    static immutable shape534 = [5, 3, 4];
+    static immutable shape45 = [4, 5];
+    static immutable shape35 = [3, 5];
+    static immutable shape34 = [3, 4];
+	
+    auto x = slice.adjTransposed;
+    assert(x.shape == shape345);
+    assert(x.front.shape == shape45);
+    assert(x.front == iota([4, 5]));
+    x.popFront;
+    assert(x.front == iota([4, 5], (4 * 5)));
+	
+    auto y = slice.adjTransposed!1;
+    assert(y.shape == shape435);
+    assert(y.front.shape == shape35);
+    int err;
+    assert(y.front == slice.universal.strided!1(4).reshape([3, -1], err));
+    y.popFront;
+    assert(y.front.front == iota([5], 5));
+	
+    auto z = slice.adjTransposed!2;
+    assert(z.shape == shape534);
+    assert(z.front.shape == shape34);
+    assert(z.front == slice.universal.strided!2(5).reshape([3, -1], err));
+    z.popFront;
+    assert(z.front.front == iota([4], 1, 5));
+}
+
+// Ensure works on canonical
+@safe @nogc pure nothrow
+unittest
+{
+    import mir.ndslice : iota, canonical;
+	
+    auto slice = iota(3, 4).canonical;
+	
+    static immutable shape34 = [3, 4];
+    static immutable shape43 = [4, 3];
+	
+    auto x = slice.adjTransposed;
+    assert(x.shape == shape34);
+    assert(x.front == iota(4));
+    x.popFront;
+    assert(x.front == iota([4], 4));
+	
+    auto y = slice.adjTransposed!1;
+    assert(y.shape == shape43);
+    assert(y.front == iota([3], 0, 4));
+    y.popFront;
+    assert(y.front == iota([3], 1, 4));
+}
+
+// Ensure works on universal
+@safe @nogc pure nothrow
+unittest
+{
+    import mir.ndslice : iota, universal;
+	
+    auto slice = iota(3, 4).universal;
+	
+    static immutable shape34 = [3, 4];
+    static immutable shape43 = [4, 3];
+	
+    auto x = slice.adjTransposed;
+    assert(x.shape == shape34);
+    assert(x.front == iota(4));
+    x.popFront;
+    assert(x.front == iota([4], 4));
+	
+    auto y = slice.adjTransposed!1;
+    assert(y.shape == shape43);
+    assert(y.front == iota([3], 0, 4));
+    y.popFront;
+    assert(y.front == iota([3], 1, 4));
+}
+
+/++
+Convenience function to return a slice that can be iterated by dimension. 
+
+Combines $(LREF transposed) and $(LREF ipack) together.
+
+Params:
+    dim = dimension to perform iteration on
+    slice = input slice (may not be 1-dimensional slice)
+Returns:
+    n-dimensional slice ipacked to allow iteration by dimension
+See_also:
+	$(LREF slice),
+	$(LREF ipack),
+	$(LREF transposed).
++/
+template byDim(size_t dim = 0)
+{
+    import mir.ndslice : Slice, SliceKind;
+	
+    auto byDim(SliceKind kind, size_t[] packs, Iterator)(Slice!(kind, packs, Iterator) slice)
+    {
+        static assert (packs != [1], "1-dimensional slice not accepted");
+		
+        import mir.ndslice : ipack;
+        return slice.adjTransposed!dim.ipack!1;
+    }
+}
+
+/// 2-dimensional slice support
+@safe @nogc pure nothrow
+unittest
+{
+    import mir.ndslice : iota;
+	
+    auto slice = iota(3, 4);
```

Comment:
please add comments to `iota` all iota declarations in documented unittests.
like this 
```d
//  0  1  2  3
//  4  5  6  7
//  8  9 10 11
``` 
This will be very helpful for all newcomers